src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -197,8 +199,8 @@ public static String mkNameForPath(List<Field> currentField) {
      * @throws NonNullableFieldWasNullException in the {@link CompletableFuture} if a non-null field resolved to a null value
      */
     @SuppressWarnings("unchecked")
-    protected Object /* CompletableFuture<Map<String, Object>> | Map<String, Object> */
-    executeObject(ExecutionContext executionContext, ExecutionStrategyParameters parameters) throws NonNullableFieldWasNullException {
+    @DuckTyped(shape = "CompletableFuture<Map<String, Object>> | Map<String, Object>")
```

Comment:
I think this annotation reads better in the IDEA rather than a comment.  Maybe we can use it later in some way but honestly its just a more type safe way of saying the same as we try to say in the comment

Reply:
Nice idea. Makes it more official than a comment